internal/eval/eval.go

```diff
@@ -2227,6 +2227,29 @@ func evalLPOP(args []string, store *dstore.Store) []byte {
 	return clientio.Encode(x, false)
 }
 
+func evalLLEN(args []string, store *dstore.Store) []byte {
+	if len(args) != 1 {
+		return diceerrors.NewErrArity("LLEN")
+	}
+
+	obj := store.Get(args[0])
+	if obj == nil {
+		return clientio.Encode(0, false)
+	}
+
+	// if object is a set type, return error
+	if object.AssertType(obj.TypeEncoding, object.ObjTypeSet) == nil {
+		return diceerrors.NewErrWithFormattedMessage(diceerrors.WrongTypeErr)
+	}
+
+	if err := object.AssertTypeAndEncoding(obj.TypeEncoding, object.ObjTypeByteList, object.ObjEncodingDeque); err != nil {
+		return clientio.Encode(err, false)
```

Comment:
Return the error directly, it is already encoded. Include a test case validating the error message too.
```suggestion
		return err

Reply:
Done